php/login.php

```diff
@@ -1,103 +1,110 @@
-    $mypassword = mysqli_real_escape_string( $db,$_POST['password'] );
-
-    $sql = "SELECT ID FROM accounts WHERE username = '$myusername' and password = SHA2('$mypassword', 512)";
-    $result = mysqli_query( $db,$sql );
-    $row = mysqli_fetch_array( $result );
-    $active = $row['active'];
-
-    $count = mysqli_num_rows( $result );
-
-    // If result matched $myusername and $mypassword, table row must be 1 row
-
-    if( $count == 1 )
-    {
-        session_register( "myusername" );
-        $_SESSION['login_user'] = $myusername;
-
-        redirect( "welcome.php" );
-        exit();
+// Include config file
+require_once 'config.php';
+ 
+// Define variables and initialize with empty values
+$username = $password = "";
+$username_err = $password_err = "";
+ 
+// Processing form data when form is submitted
+if($_SERVER["REQUEST_METHOD"] == "POST"){
+ 
+    // Check if username is empty
+    if(empty(trim($_POST["username"]))){
+        $username_err = 'Please enter username.';
+    } else{
+        $username = trim($_POST["username"]);
     }
-    else
-    {
-        $error = "ERROR: Your Login Name or Password is invalid<br>POST USER: ".$_POST['username']."<br>POST PASSWORD: ".$_POST['password']."<br>myusername: ".$myusername."<br>mypassword: ".$mypassword."<br> Result Count: ".$count;
+    
+    // Check if password is empty
+    if(empty(trim($_POST['password']))){
+        $password_err = 'Please enter your password.';
+    } else{
+        $password = trim($_POST['password']);
     }
+    
+    // Validate credentials
+    if(empty($username_err) && empty($password_err)){
+        // Prepare a select statement
+        $sql = "SELECT username, password FROM accounts WHERE username = ?";
+        
+        if($stmt = mysqli_prepare($link, $sql)){
+            // Bind variables to the prepared statement as parameters
+            mysqli_stmt_bind_param($stmt, "s", $param_username);
+            
+            // Set parameters
+            $param_username = $username;
+            
+            // Attempt to execute the prepared statement
+            if(mysqli_stmt_execute($stmt)){
+                // Store result
+                mysqli_stmt_store_result($stmt);
+                
+                // Check if username exists, if yes then verify password
+                if(mysqli_stmt_num_rows($stmt) == 1){                    
+                    // Bind result variables
+                    mysqli_stmt_bind_result($stmt, $username, $hashed_password);
+                    if(mysqli_stmt_fetch($stmt)){
+                        if($password == $hashed_password){
+                            /* Password is correct, so start a new session and
+                            save the username to the session */
+                            session_start();
+                            $_SESSION['username'] = $username;      
+                            header("location: welcome.php");
```

Comment:
Verified password issue with Firefox, Explorer, Edge (unable to test with Chrome, and Safari however I'm sure results will be the same)
No error message is being returned and password is not being accepted. See following image: ![Screen capture](https://i.imgur.com/aniXmu0.png)